src/start/RealmInputScreen.js

```diff
@@ -58,6 +86,46 @@ const tryParseInput = (realmInputValue: string): MaybeParsedInput => {
   return { valid: true, value: url };
 };
 
+type Suggestion =
+  | ValidationError // Display relevant validation error message
+  | string // Suggest this string as the server URL
+  | null; // No suggestion
+
+function getSuggestion(realmInputValue, maybeParsedInput): Suggestion {
+  if (
+    !maybeParsedInput.valid
+    // Ignore other errors, like InvalidUrl, which often happen when the
+    // user just hasn't finished typing a good URL. Those errors will still
+    // show up if they apply at submit time; see the submit handler.
+    && (maybeParsedInput.error === ValidationError.NoUseEmail
+      || maybeParsedInput.error === ValidationError.UnsupportedSchemeZulip
+      || maybeParsedInput.error === ValidationError.UnsupportedSchemeOther)
```

Comment:
This logic makes sense. Let's express it with a `switch`, exhaustively — that way it's explicit what errors we *are* allowing through to the logic below, and also when adding new errors that will make it much more likely that we make an intentional choice of which way they go here.